src/verify_cert.rs

```diff
@@ -668,44 +665,49 @@ mod tests {
         ));
     }
 
+    #[test]
     #[cfg(feature = "alloc")]
-    fn build_linear_chain(chain_length: usize) -> Result<(), ControlFlow<Error, Error>> {
-        let ca_cert = make_issuer(format!("Bogus Subject {chain_length}"), None);
+    fn longest_allowed_path() {
+        for len in 1..=6 {
+            let ca_cert = make_issuer(format!("Bogus Anchor"), None);
+            let ca_cert_der = CertificateDer::from(ca_cert.serialize_der().unwrap());
+            let anchors = &[extract_trust_anchor(&ca_cert_der).unwrap()];
+
+            let mut intermediates = Vec::with_capacity(len);
+            let mut issuer = ca_cert;
+            for i in 0..len {
+                let intermediate = make_issuer(format!("Bogus Subject {i}"), None);
+                let intermediate_der = intermediate.serialize_der_with_signer(&issuer).unwrap();
+                intermediates.push(CertificateDer::from(intermediate_der));
+                issuer = intermediate;
+            }
+
+            let ee_der = make_end_entity(&issuer);
+            let ee_cert = EndEntityCert::try_from(&ee_der).unwrap();
+            assert!(verify_chain(anchors, &intermediates, &ee_cert, None).is_ok());
```

Comment:
it seems it would be instructive if this (and other success-case tests?) asserted that `verify_chain` returned `Ok(anchors[0])`

Reply:
Done.